sktime/series_as_features/base/estimators/interval_based/_tsf.py

```diff
@@ -159,14 +166,19 @@ def _get_intervals(n_intervals, min_interval, series_length, rng):
     return intervals
 
 
-def _fit_estimator(X, y, base_estimator, intervals, random_state=None):
+def _fit_estimator(estimator, X, y, intervals):
     """
-    Fit an estimator - a clone of base_estimator - on input data (X, y)
-    transformed using the randomly generated intervals.
+    Fit an estimator on input data (X, y) transformed using
+    the randomly generated intervals.
     """
+    transformed_x = _transform(X, intervals)
+    return estimator.fit(transformed_x, y)
 
+
+def _clone_estimator(base_estimator, random_state=None):
```

Comment:
Could we move this to utils here or in PR #796?

Reply:
I was thinking about doing this in #796 but now I don't know which of these PRs have a chance to be merged first (I need to update one of them, merge changes into a seconds one and, only after that, rework the code).

Reply:
Alright makes sense, let's leave it as it is and we see which one gets merged first

Reply:
@kachayev now that #796 is merged, I can merge this one once you move the function to utils. Thanks!

Reply:
@mloning on it!

Reply:
@kachayev the failing linux builds on main are now fixed, happy to merge this one once the function is moved to utils and reused in the other code